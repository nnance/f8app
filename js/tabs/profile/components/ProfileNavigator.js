import React from 'react';
import FixBugScrollViewNavigator from '../../../common/FixBugScrollViewNavigator';

import FollowerScreen from '../containers/FollowerScreen';
import FollowingScreen from '../containers/FollowingScreen';
import MyFanScreen from '../containers/MyFanScreen';
import ActivityScreen from '../containers/ActivityScreen';
import MyClogScreen from '../containers/MyClogScreen';
import BookmarkScreen from '../containers/BookmarkScreen';
import JellyShopScreen from '../components/JellyShopScreen';
import Home from '../containers/Home';
import ProfileEditorScreen from '../containers/ProfileEditorScreen';
import ChangeEmailScreen from '../containers/ChangeEmailScreen';
import ChangePasswordScreen from '../containers/ChangePasswordScreen';

class ProfileNavigator extends React.Component {
  constructor(...args) {
    super(...args);
    this.renderScene = this.renderScene.bind(this);
    this.onMenuPress = this.onMenuPress.bind(this);
    this.goToChangeEmail = this.goToChangeEmail.bind(this);
    this.goToChangePassword = this.goToChangePassword.bind(this);
    this.pushPage = this.pushPage.bind(this);
    this.onBack = this.onBack.bind(this);
  }

  onMenuPress(name) {
    const navigator = this.navigator;
    if (name === 'myfan') {
      navigator.push({ page: 'myfan' });
      this.props.navigate('profile/myfan');
    }
    if (name === 'activity') {
      navigator.push({ page: 'activity' });
      this.props.navigate('profile/activity');
    }
    if (name === 'myclog') {
      navigator.push({ page: 'myclog' });
      this.props.navigate('profile/myclog');
    }
    if (name === 'bookmark') {
      navigator.push({ page: 'bookmark' });
      this.props.navigate('profile/bookmark');
    }
    if (name === 'jellyShop') {
      navigator.push({ page: 'jellyShop' });
      this.props.navigate('profile/jellyShop');
    }
  }

  onBack() {
    const navigator = this.navigator;
    this.props.navigateBack();
    navigator.pop();
  }

  goToChangeEmail() {
    this.navigator.push({ page: 'change-email' });
    this.props.navigate('profile/change-email');
  }

  goToChangePassword() {
    this.navigator.push({ page: 'change-password' });
    this.props.navigate('profile/change-password');
  }

  pushPage(page) {
    this.navigator.push({ page });
    this.props.navigate(`profile/${page}`);
  }

  renderScene(route) {
    if (route.page === 'following') {
      return <FollowingScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'follower') {
      return <FollowerScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'myfan') {
      return <MyFanScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'activity') {
      return <ActivityScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'myclog') {
      return <MyClogScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'bookmark') {
      return (<BookmarkScreen
        onBackPress={this.onBack}
        redirectTo={this.props.redirectTo}
        setTabViewScrollable={this.props.setTabViewScrollable}
      />);
    }
    if (route.page === 'jellyShop') {
      return <JellyShopScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'edit-profile') {
      return (<ProfileEditorScreen
        goToChangeEmail={this.goToChangeEmail}
        goToChangePassword={this.goToChangePassword}
        onBackPress={this.onBack}
      />);
    }
    if (route.page === 'change-email') {
      return <ChangeEmailScreen onBackPress={this.onBack} />;
    }
    if (route.page === 'change-password') {
      return <ChangePasswordScreen onBackPress={this.onBack} />;
    }
    return (<Home
      onMenuPress={this.onMenuPress}
      onFollowingPress={() => this.pushPage('following')}
      onFollowerPress={() => this.pushPage('follower')}
      onEditProfile={() => this.pushPage('edit-profile')}
    />);
  }

  render() {
    return (
      <FixBugScrollViewNavigator
        ref={(node) => {
          this.navigator = node;
        }}
        initialRoute={{ page: 'home' }}
        renderScene={this.renderScene}
      />
    );
  }
}

export default ProfileNavigator;
