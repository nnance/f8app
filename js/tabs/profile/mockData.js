export const user = {
  id: 1,
  name: 'test',
  sex: 'M',
  facebookLinked: false,
};
export const followingCount = 1800;
export const followerCount = 1800;
export const following = [{ name: 'Art Nattapat' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }, { name: 'Art Art Art' }];
export const follower = [{ name: 'Art Nattapat', following: true }, { name: 'Art Art Art', following: false }, { name: 'Art Art Art', following: false }, { name: 'Art Art Art', following: true }, { name: 'Art Art Art', following: true }, { name: 'Art Art Art', following: false }];
export const myFan = [{ name: 'Art Nattapat', candys: 1250 }, { name: 'Art Art Art', candys: 1250 }, { name: 'Art Art Art', candys: 1000 }, { name: 'Art Art Art', candys: 789 }, { name: 'Art Art Art', candys: 7013 }, { name: 'Art Art Art', candys: 0 }];
export const candys = 1890;
export const activity = [{
  activity: 'like',
  title: 'แมค เดมอน ตะลุยอวกาศ',
  outline: 'The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร',
  uri: 'http://3.bp.blogspot.com/-zW6wqY_1Me0/VYvJMOV4mcI/AAAAAAAAD-4/mB_AxhFoJH4/s1600/178491main_sig07-009-516.jpg',
  date: new Date(2016, 10, 4),
}, {
  activity: 'like',
  title: 'แมค เดมอน ตะลุยอวกาศ',
  outline: 'The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร',
  uri: 'http://3.bp.blogspot.com/-zW6wqY_1Me0/VYvJMOV4mcI/AAAAAAAAD-4/mB_AxhFoJH4/s1600/178491main_sig07-009-516.jpg',
  date: new Date(2016, 10, 8),
}, {
  activity: 'read',
  title: 'แมค เดมอน ตะลุยอวกาศ',
  outline: 'The Martain กำกับโดย ริดลีย์ สก๊อต เนื้อหาเล่าถึง นักบินอวกาศที่ถูกทิ้งไว้บนดาวอังคาร',
  uri: 'http://www.fujisan.ne.jp/fit/th/images/ohishikouen430.jpg',
  date: new Date(2015, 10, 8),
}];

export const myClogs = [
  {
    title: 'Richy Rich! รวยมากนะ! รู้ยังคะทุกคน',
    author: {
      name: 'David Beckham',
    },
    preview: require('../../assets/A.png'),
    category: 'M',
    viewCount: 12300,
    likeCount: 1500,
    createdAt: new Date(2015, 5, 24),
  },
  {
    title: 'Money Honey คุณชายหน้าตายกับยัยขี้งก',
    author: {
      name: 'สุดสาครนอย ไทรโยค',
    },
    preview: require('../../assets/B.png'),
    category: 'N',
    viewCount: 100,
    likeCount: 55,
    createdAt: new Date(2016, 10, 8),
  },
];

export const bookmark = [
  {
    title: 'Richy Rich! รวยมากนะ! รู้ยังคะทุกคน',
    preview: require('../../assets/A.png'),
    category: 'N',
    bookmarkCount: 7,
  },
  {
    title: 'Money Honey คุณชายหน้าตายกับยัยขี้งก',
    preview: require('../../assets/B.png'),
    category: 'M',
    bookmarkCount: 3,
  },
];
