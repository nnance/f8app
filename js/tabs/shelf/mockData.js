export const topClog = {
  title: 'แมค เดมอน ตะลุยอวกาศ',
  author: 'Steve Jobs',
  review: 'The martin กำลัง บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ บลาๆๆๆๆๆ',
};

export const fakeMetaClog = [
  {
    title: 'Mr.Smith Goes to somewhere',
    author: 'David Beckham',
  },
  {
    title: 'Mr.Smith Goes to somewhere',
    author: 'David Beckham2',
  },
  {
    title: 'Mr.Smith Goes to somewhere',
    author: 'David Beckham3',
  },
  {
    title: 'Mr.Smith Goes to somewhere',
    author: 'David Beckham4',
  },
  {
    title: 'Mr.Smith Goes to somewhere',
    author: 'David Beckham5',
  },
  {
    title: 'Mr.Smith Goes to somewhere',
    author: 'David Beckham6',
  },
];

export const highlightCategory = [
  {
    title: 'DRAMATIC SERIES',
    following: false,
    clogs: fakeMetaClog,
  },
  {
    title: 'NEW FUNNY CLOG',
    following: false,
    clogs: fakeMetaClog,
  },
  {
    title: 'FANTASY SCI-FI',
    following: true,
    clogs: fakeMetaClog,
  },
];
