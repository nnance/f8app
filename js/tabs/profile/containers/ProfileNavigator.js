import { withTracking } from '../../../common/navigateTracking';

import ProfileNavigator from '../components/ProfileNavigator';

export default withTracking(ProfileNavigator);
