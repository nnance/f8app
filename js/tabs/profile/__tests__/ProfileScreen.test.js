import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { View } from 'react-native';

import ProfileNavigator from '../components/ProfileNavigator';
import * as mockData from '../mockData';

describe('ProfileNavigator', () => {
  it('renderScene correct component', () => {
    const scenes = ['following', 'follower', 'myfan', 'activity', 'myclog', 'bookmark', 'jellyShop', 'edit-profile', 'change-email', 'change-password', undefined];
    const dump = shallow(<ProfileNavigator {...mockData} />);
    scenes.forEach((scene) => {
      const component = dump.instance().renderScene({ page: scene });
      const tree = shallow(<View>{component}</View>);
      expect(toJSON(tree)).toMatchSnapshot();
    });
  });

  it('onMenuPress, should push into navigator', () => {
    const dump = shallow(<ProfileNavigator {...mockData} navigate={jest.fn()} />);
    const menues = ['myfan', 'activity', 'myclog', 'bookmark', 'jellyShop'];
    menues.forEach((menu) => {
      const spy = jest.fn();
      dump.instance().navigator = {
        push: spy,
      };
      dump.instance().onMenuPress(menu);
      expect(spy).toBeCalledWith({ page: menu });
    });
  });

  it('onBack', () => {
    const spy = jest.fn();
    const dump = shallow(<ProfileNavigator {...mockData} navigateBack={jest.fn()} />);
    dump.instance().navigator = {
      pop: spy,
    };
    dump.instance().onBack();
    expect(spy).toBeCalled();
  });

  it('pushPage', () => {
    const spy = jest.fn();
    const dump = shallow(<ProfileNavigator {...mockData} navigate={jest.fn()} />);
    dump.instance().navigator = {
      push: spy,
    };
    dump.instance().pushPage('test');
    expect(spy).toBeCalledWith({ page: 'test' });
  });
});
