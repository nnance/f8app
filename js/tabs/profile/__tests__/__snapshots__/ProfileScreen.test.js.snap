exports[`ProfileNavigator renderScene correct component 1`] = `
<View>
  <Connect(FollowingScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 2`] = `
<View>
  <Connect(FollowerScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 3`] = `
<View>
  <Connect(MyFanScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 4`] = `
<View>
  <Connect(ActivityScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 5`] = `
<View>
  <Connect(MyClogScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 6`] = `
<View>
  <Apollo(Apollo(BookmarkScreen))
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 7`] = `
<View>
  <JellyShopScreen
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 8`] = `
<View>
  <Connect(ProfileEditorScreen)
    goToChangeEmail={[Function]}
    goToChangePassword={[Function]}
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 9`] = `
<View>
  <Connect(ChangeEmailScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 10`] = `
<View>
  <Connect(ChangePasswordScreen)
    onBackPress={[Function]} />
</View>
`;

exports[`ProfileNavigator renderScene correct component 11`] = `
<View>
  <Connect(Home)
    onEditProfile={[Function]}
    onFollowerPress={[Function]}
    onFollowingPress={[Function]}
    onMenuPress={[Function]} />
</View>
`;
